prow/plugins/trigger/trigger.go

```diff
@@ -316,6 +316,16 @@ func RunRequestedWithLabels(c Client, pr *github.PullRequest, baseSHA string, re
 
 func runRequested(c Client, pr *github.PullRequest, baseSHA string, requestedJobs []config.Presubmit, eventGUID string, labels map[string]string, millisecondOverride ...time.Duration) error {
 	var errors []error
+
+	// If the PR is not mergeable, we should not run any jobs.
+	// This is to prevent the case where a PR is not mergeable, but the user
+	// still wants to run tests by pushing or etc. This can happen if the PR is in the process of
+	// being rebased, or if the PR is in a bad state.
+	if pr.Mergable != nil && !*pr.Mergable {
+		c.Logger.Warn("PR is not mergeable, skipping all jobs.")
```

Comment:
This won't be the ideal case for the user. I prefer to leave a comment explaining why we won't trigger any jobs.

Reply:
Thanks for your time in reviewing my changes, I changed the comment to explain why we don't run jobs when PR is not mergeable.

Reply:
I am not sure if this is the correct approach. We definitely need to run the job of Draft PRs since people are using this concept already. Is there anything specific in the pull request to determine whether it is conflicted? Also, about the message, I mean to provide a message to the user directly in the PR, otherwise, this will lead to confusion and people will start asking why their jobs are not being triggered.

Reply:
Got the idea, you are right, it's not a good idea to do this for all non-mergeable PR, let me investigate more about it and write a better message.

Reply:
@droslean I've found we have a draft variable to determine whether the PR is in draft or not.

Reply:
IMHO we should not care about the draft PRs. We only shouldn't trigger the jobs if the PR needs rebase. The `needs-rebase` plugin will post a comment in the PR to inform the user, so maybe it is not necessary for another comment or a specific comment that explains why the jobs won't trigger would be nice.

Reply:
Maybe it would be better to check labels then https://github.com/kubernetes/test-infra/blob/d8035cceb460e2f65888a6c2c7afa0c77af09796/prow/github/types.go#L250 https://github.com/kubernetes/test-infra/blob/d8035cceb460e2f65888a6c2c7afa0c77af09796/prow/labels/labels.go#L46 instead of the mergeable field since we know it's a possibility the field may not be computed yet at that moment

Reply:
There is no reason to make this more complicated than it is. Let's just skip triggering the jobs if the PR is not mergeable, but let's first make sure that GH doesn't treat the Draft PRs as non-mergeable as well.

Reply:
It wouldn't affect on since one step before we will skip it if it's a draft. https://github.com/kubernetes/test-infra/blob/master/prow/plugins/trigger/pull-request.go#L342
I'm not sure but the draft has another field in the struct and I think it will haven't relation to the mergeable field